src/opnsense/www/js/opnsense_bootgrid.js

```diff
@@ -1184,6 +1211,41 @@ class UIBootgrid {
         this.table.on('dataProcessed', expandOnceAfterReload);
     }
 
+    /**
+     * Expand top level dataTree rows whose immediate child count increased.
+     *
+     * In tree view, category labels are not unique and the new/updated row
+     * could appear under any matching label or even create a new top-level node.
+     * Tracking by row index before/after reload ensures only the actual
+     * parent row(s) whose child count grew are expanded, avoiding expanding
+     * all rows with the same label.
+     */
+    expandDataTree() {
+        if (!this.table?.options?.dataTree) return;
+
+        const snapshot = () => {
+            const rowMap = new Map();
+            (this.table.getRows?.() || []).forEach(row => {
+                rowMap.set(row.getIndex?.(), (row.getTreeChildren?.() || []).length);
+            });
+            return rowMap;
+        };
+
+        const beforeSnapshot = snapshot();
+
+        const expandOnceAfterReload = () => {
+            this.table.off('dataProcessed', expandOnceAfterReload);
+            const afterSnapshot = snapshot();
+            afterSnapshot.forEach((childCount, rowId) => {
+                if (childCount > (beforeSnapshot.get(rowId) || 0)) {
```

Comment:
Want to quickly point out that while this code is valid, these types of constructs are often better dealt with through null coalescing (`??`). If the fallback was supposed to be `|| 42`, this would always overwrite an intended`0` inserted into the Map.

No problem here but just general feedback :)